src/vncproxy.ml

```diff
@@ -121,43 +119,46 @@ let bind_local_port () =
 
 let connect_to_console session_id console =
   let uri = Uri.of_string console.API.console_location in
-  match%lwt Xen_api_lwt_unix.Lwt_unix_IO.open_connection uri with
+  Xen_api_lwt_unix.Lwt_unix_IO.open_connection uri >>= function
```

Comment:
We were not having this in a `Lwt.catch` because we use the exit to cleanup the resources?

Reply:
Yes, it looks like the case reading the main function below.